### ISSUE TITLE
Guess email domain for corresponding author from institutional info

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#1049]
Once we have the institutional info for someone, it would good makes guesses of the email and verify them. If we see a positive response to those guesses, we should provide them as suggestions or notify the admin that there was no match. 

To make the guesses, we could:
* look up email domains from the institution. There are datasets for this that we've already worked with Mark. Hunter.io provides this too, which may be the best bet. 
* ping the domains to hunter.io, which will give us the structure e.g {first}_{last}@domain.edu.
* Slot info we know into the email, e.g if we know the author's name from our scrape. 

To verify the guesses, we could:

* hit the hunter.io domain, or maybe the mailgun service verifier. 
* if it passes, provide the guess to the admin. 

Open to better routes on those if you have them @markmacgillivray, only thought about this briefly, but seems reasonable.

**Comment by markmacgillivray** [comment]
Once we create a request and the user saves it, if we don't yet have an email address we do a page scrape. If that does not find the email, then if we have run a crossref search on it, we go through the authors provided from crossref. If any of those have affiliation, it is sent to Hunter as a company name along with the author first name and last name. If this returns an email, then that is used as the email for the request. Note though that affiliations straight from crossref are unlikely to match as they are descriptive strings rather than official names.

There does not appear to be anywhere reliable that we get a domain for an author. Also, it is not clear if Hunter would do any better with a domain than it does with a proper organisational name anyway - and getting the proper organisational name seems to be the harder part, or at least something that is likely to be found along with a domain. Hence I have focused on using the domain.

If a request is subsequently saved, e.g. by the admin, then if there is still not an email address but there is an affiliation, then we check with Hunter again. So at this stage, if the admin is able to change the author affiliation (it iseditable in the admin area now) then Hunter may be more likely to find it.

As Hunter only provides valid email addresses anyway, I have not bothered validating it against the Hunter or Mailgun email validation APIs.

This would be hard to test as there are only a few DOIs that would return author affiliations, and only a few of those that would return usable affiliation strings that Hunter would match, and then only some of those that would return an email address for the user first and last names in combination with that organisational name.

So, let me know if you want to try testing this to see how many hits it gets, or if you are happy just to put it onto live (it certainly doesn't do anything worse than we used to, so no negative impact).

**Comment by JosephMcArthur** [comment]
lets put it on live

**Comment by JosephMcArthur** [comment]
close if needed Mark. cc @natalianorori for info here. 

Am I right in thinking that if Natalia updates the affiliation, this will automatically try and find the email for her on save. So, if we don't have the email it makes sense to try and find the institution. If that doesn't work, it makes sense to try and do the normal email finding?

**Comment by markmacgillivray** [comment]
Yes, if we have authors from crossref then the first author in their list is the corresponding author. We use their name and email address if we can find one, and we use their affiliation too, but the affiliations can be strings that are not very useful. So if @natalianorori changed the affiliation and saved the request, and it at that point still does not have an author email address, it would try to use hunter to get an email address for the first author first and last name along with the affiliation value Natalia just provided. If it finds an email, it will add that to the request.

**Comment by JosephMcArthur** [comment]
Bit of a realization here on a call with @natalianorori, we can often find the domain of the author from Scholar or similar. We can shortcut the guessing process outlined above if we had a field for admins to directly feed in the domain. I think this is worth adding. It can also be filled in where the guess from institution comes back with something, which will also make that more transparent. If the field is filled by an admin, that should cause a recheck of the email via hunter.
